local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -213,10 +213,8 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 		pvName := generatePVName(file, d.Node.Name, class)
 		pv, exists := d.Cache.GetPV(pvName)
 		if exists {
-			if volMode == v1.PersistentVolumeBlock && (pv.Spec.VolumeMode == nil ||
-				*pv.Spec.VolumeMode != v1.PersistentVolumeBlock) {
-				errStr := fmt.Sprintf("Incorrect Volume Mode: PV %q (path %q) was not created in block mode. "+
-					"Please check if BlockVolume features gate has been enabled for the cluster.", pvName, filePath)
+			if volMode == v1.PersistentVolumeFilesystem && *pv.Spec.VolumeMode == v1.PersistentVolumeBlock {
```

Comment:
Block volume feature is still alpha, so pv.spec.VolumeMode should be nil in non-alpha clusters.